syntax = "proto3";
package GameService;


// --------------------------
message ReadyRequest{
    bool ready = 1;
}

message ReadyResponse{
    int32 id = 1;
    bytes map = 2;
}


// --------------------------
message StepRequest{
    int32 x = 1;
    int32 y = 2;
    int32 id = 3;
}

message StepResponse{
    bool win = 1;
    bytes map = 2;
}

// --------------------------
message StateRequest{
    int32 id = 1;
}

message StateResponse{    
    bytes map = 1;
}


// --------------------------
service GameService{
    rpc StartGame(ReadyRequest) returns (ReadyResponse);
    rpc MakeStep(StepRequest) returns (StepResponse);
    rpc GetState(StateRequest) returns (StateResponse);
}